pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -886,43 +932,30 @@ func (os *OpenStack) getVolume(diskName string) (volumes.Volume, error) {
 	return volume, err
 }
 
-func (os *OpenStack) getComputeIDbyHostname(cClient *gophercloud.ServiceClient) (string, error) {
+// getLocalComputeID tries to find ID of the OpenStack instance where
+// this code runs.
+func (os *OpenStack) getLocalComputeID(cClient *gophercloud.ServiceClient) (string, error) {
+	// try to read /var/lib/cloud/data/instance-id
+	const instanceIDFile = "/var/lib/cloud/data/instance-id"
+	idBytes, err := ioutil.ReadFile(instanceIDFile)
+	if err == nil {
+		id := string(idBytes)
+		id = strings.TrimSpace(id)
```

Comment:
An invalid value would at least show up in a reasonable way in errors later (if the value is "bogus", we'd see an error trying to find the "bogus" server in openstack). If this file is present, but contains no value, I think this should probably return an error